### PR TITLE
Feature/CMRARC-544

### PR DESCRIPTION
**Pull request by ryanmiller-1**
This also covers CMRARC-545 and CMRARC-550.

- Removed files associated with old e-mail system.  They do not provide a good basis for future e-mails and are not currently being used.  This includes references to them in models/controllers, previews, and tests.
- Updated Puma because a minor version was released that satisfies dependabot.
- Migrated 'daac' column out of record_data into records table and updated associated test fixtures and ingest procedures
- added the ability for users to specify multiple different e-mail timing preferences (CMRARC-550).
- added formatting for digest e-mails (largely taken from MMT)
- added a cron and helper for generating these e-mails (CMRARC-545)
- updated environment files to be able to send e-mails (and with correct URLs)

### CODE REVIEW

File: app/views/layouts/mailer.html.erb [hunk 1]
```diff
@@ -1,12 +1,182 @@
...
+      table {
+        width: 100%;
+        text-align: left;
+      }
+
+      thead th {
+        background-color: #cbcbcb
+      }
+
+      /***************************************************
```
Comment (ryanmiller-1) [inline app/views/layouts/mailer.html.erb]: rehome these

File: lib/rake_helpers/daac_curator_mailer_helper.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class DaacCuratorMailerOrchestration
+  def self.released_records_digest_conductor(now: Time.now)
+    emails_sent = 0
+    current_recipients = ['daily']
+    # wday returns 0-6 where 1 is Monday
+    current_recipients << 'weekly' if now.wday == 1
+    current_recipients << 'biweekly' if now.wday == 1 && ((1..7).to_a.include?(now.day) || (15..21).to_a.include?(now.day))
+    current_recipients << 'monthly' if now.wday == 1 && (1..7).to_a.include?(now.day)
```
Comment (ryanmiller-1) [inline lib/rake_helpers/daac_curator_mailer_helper.rb]: rename current_recipients

File: app/models/collection.rb [hunk 3]
```diff
@@ -37,17 +37,15 @@ def self.create_record(concept_id, revision_id, data_format, collection_data, op
     Collection.transaction do
       collection = Collection.find_or_create_by(concept_id: concept_id)
       collection.update_attributes!(short_name: short_name)
-
-      new_record = Record.create(recordable: collection, revision_id: revision_id, format: data_format)
-
       daac = concept_id.split('-').last
```
[the remark this answers is not in the document]
Comment (ryanmiller-1) [reply app/models/collection.rb]: Is this resolved?

[the remark this answers is not in the document]
Comment (charleshuang80) [reply app/models/collection.rb]: yup

**Pull request merged**